zeebe/gateway-protocol/src/main/proto/rest-api.yaml

```diff
@@ -1706,26 +1721,45 @@ components:
           format: int64
         tenantId:
           type: string
-    AuthorizationAssignRequest:
+    AuthorizationPatchRequest:
       type: object
       properties:
-        ownerKey:
-          type: integer
-          format: int64
-        ownerType:
+        action:
+          description: Indicates if permissions should be added or removed.
           type: string
           enum:
+            - ADD
+            - REMOVE
+        resourceType:
+          description: The type of resource to add/remove perissions to/from.
+          enum:
+            - AUTHORIZATION
+            - MESSAGE
+            - JOB
+            - APPLICATION
+            - TENANT
+            - DEPLOYMENT
+            - PROCESS_DEFINITION
+            - USER_TASK
+            - DECISION_REQUIREMENTS_DEFINITION
+            - DECISION_DEFINITION
+            - USER_GROUP
             - USER
             - ROLE
-            - GROUP
-        resourceKey:
-          type: string
-        resourceType:
-          type: string
         permissions:
           type: array
           items:
-            type: string
+            properties:
+              permissionType:
+                enum:
+                  - CREATE
+                  - READ
+                  - UPDATE
+                  - DELETE
+              resourceIds:
```

Comment:
❓ This is something that we discussed in the kick off and I'm still not quite sure about how we phrase this, is it Ids, are they keys? Should we have a term that is different to either of them? I'm not sure first had of the usage of IDs and Keys in the system so it may be fine that ID is used here.

Reply:
The conclusion in the kickoff was that we should Ids instead of keys. A key is a long value that's generated within Zeebe. An Id is generally a String identifier.